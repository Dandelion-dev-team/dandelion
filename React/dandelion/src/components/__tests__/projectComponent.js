import React from "react"
import renderer from "react-test-renderer"
import ProjectComponent from "../projectComponent"


describe("Project", () => {
    let originalFetch;

    beforeEach(() => {
        originalFetch = global.fetch;
        global.fetch = jest.fn(() => Promise.resolve({
            json: () => Promise.resolve({
                "projects": [
                    {
                      "id": 0,
                      "title": "Lemon Balm",
                      "owner": "GLA42012",
                      "type": "Internal",
                      "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.",
                      "project_image_link": "https://cdn.britannica.com/77/124177-004-FEB6DB88/Balm-lemon-balm.jpg",
                      "project_text": "string",
                      "start_date": "2022-01-31T14:31:24.262Z",
                      "end_date": "2022-01-31T14:31:24.262Z",
                      "status": "active"
                    }
                ]
            })
        }));
    });

    afterEach(() => {
        global.fetch = originalFetch;
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(<ProjectComponent/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

