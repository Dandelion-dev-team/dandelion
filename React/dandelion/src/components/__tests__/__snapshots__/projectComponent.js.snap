// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders correctly 1`] = `
<div
  className="project-table"
>
  <table
    className="projectList"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th>
          Owner
        </th>
        <th>
          Type
        </th>
      </tr>
    </thead>
  </table>
  <div
    className="add-btn"
  >
    <button
      className="submitButton"
    >
      Add Project
    </button>
  </div>
</div>
`;
